pydantic/generics.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, Generic, NoReturn, Tuple, Type, TypeVar, Union, get_type_hints
+
+from pydantic import BaseModel, create_model
+from pydantic.class_validators import gather_validators
+
+_generic_types_cache: Dict[Tuple[Type[Any], Union[Any, Tuple[Any, ...]]], Type[BaseModel]] = {}
+GenericModelT = TypeVar('GenericModelT', bound='GenericModel')
+
+
+class GenericModel(BaseModel):
+    __slots__ = ()
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> NoReturn:
+        if cls is GenericModel:
```

Comment:
Does it mean it is not abstract and still can not be initialized without subclassing and not useful without implenting the function by users themselves?